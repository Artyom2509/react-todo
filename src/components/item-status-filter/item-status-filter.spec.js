import { render, shallow } from 'enzyme';
import React from 'react';
import { ItemStatusFilter } from './item-status-filter';

let store;

describe('render', () => {
	it('snapshot', () => {
		store = { filter: 'all', onFilterChange: () => {} };
		const wrapper = render(<ItemStatusFilter store={store} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should correct buttons length', () => {
		store = { filter: 'all', onFilterChange: () => {} };
		const wrapper = render(<ItemStatusFilter store={store} />);
		expect(wrapper.children('button').length).toBe(3);
	});
	it('should add button active className', () => {
		store = { filter: 'active', onFilterChange: () => {} };
		const wrapper = shallow(<ItemStatusFilter store={store} />);
		expect(wrapper.childAt(1).hasClass('btn-info')).toBeTruthy();
	});
});

describe('events', () => {
	it('should call onFilterChange', () => {
		const onFilterChange = jest.fn();
		store = { filter: 'all', onFilterChange };
		const wrapper = shallow(<ItemStatusFilter store={store} />);
		wrapper.find('.btn-info').simulate('click');
		expect(onFilterChange).toBeCalled();
	});
});
