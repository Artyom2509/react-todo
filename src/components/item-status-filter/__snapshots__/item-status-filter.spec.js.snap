// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "btn-group",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "btn btn-info",
          "type": "button",
        },
        "children": Array [
          Node {
            "data": "All",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "btn btn-outline-secondary",
            "type": "button",
          },
          "children": Array [
            Node {
              "data": "Active",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "btn btn-outline-secondary",
              "type": "button",
            },
            "children": Array [
              Node {
                "data": "Done",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "type": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "type": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "type": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "btn btn-outline-secondary",
          "type": "button",
        },
        "children": Array [
          Node {
            "data": "Active",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "btn btn-outline-secondary",
            "type": "button",
          },
          "children": Array [
            Node {
              "data": "Done",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "type": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "btn btn-info",
            "type": "button",
          },
          "children": Array [
            Node {
              "data": "All",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "type": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "type": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "btn btn-outline-secondary",
          "type": "button",
        },
        "children": Array [
          Node {
            "data": "Done",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "btn btn-outline-secondary",
            "type": "button",
          },
          "children": Array [
            Node {
              "data": "Active",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "class": "btn btn-info",
              "type": "button",
            },
            "children": Array [
              Node {
                "data": "All",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "type": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "type": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "type": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
